src/detect-lua.c

```diff
@@ -1119,6 +1120,8 @@ static void DetectLuaFree(void *ptr)
         if (lua->filename)
             SCFree(lua->filename);
 
+        DetectUnregisterThreadCtxFuncs(de_ctx, NULL, lua, "lua");
```

Comment:
Why is this part of this commit? It should be in its own. A big API update commit like this should be as clean as possible as it's hard to review before merge, but also later when someone might try to analyze why something was changed.

Reply:
I'll move it into a separate commit. I had not done so yet because this was the sole reason for the added function and the unregistration function was in the previous commit.